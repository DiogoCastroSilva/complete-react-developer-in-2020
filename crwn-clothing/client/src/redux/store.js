import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { persistStore } from 'redux-persist';
// import thunk from 'redux-thunk';
import createSagaMiddleware from 'redux-saga';

import rootReducer from './root-reducer';
import rootSaga from './root-saga';


const sagaMiddleware = createSagaMiddleware();

const middleware = [sagaMiddleware];

if (process.env.NODE_ENV === 'development') {
    middleware.push(logger);
}

export const store = createStore(
    rootReducer,
    applyMiddleware(...middleware)
);

sagaMiddleware.run(rootSaga);


export const persistor = persistStore(store);

export default { store, persistor };