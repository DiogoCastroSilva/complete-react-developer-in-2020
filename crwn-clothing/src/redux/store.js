import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { persistStore } from 'redux-persist';
// import thunk from 'redux-thunk';
import createSagaMiddleware from 'redux-saga';

import rootReducer from './root-reducer';

import { fetchCollectionsStart } from './shop/shop.sagas';

const sagaMiddleware = createSagaMiddleware();

const middleware = [sagaMiddleware];

if (process.env.NODE_ENV === 'development') {
    middleware.push(logger);
}

export const store = createStore(
    rootReducer,
    applyMiddleware(...middleware)
);

sagaMiddleware.run(fetchCollectionsStart);

export const persistor = persistStore(store);

export default { store, persistor };