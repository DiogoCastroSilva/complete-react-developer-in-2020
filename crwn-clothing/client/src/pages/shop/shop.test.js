import React from 'react';
import { mount } from 'enzyme';

import Shop from './shop.component';
import { BrowserRouter } from 'react-router-dom';
import { createStore, combineReducers } from 'redux';
import { Provider } from 'react-redux';


export const createMockStore = ({ state, reducers }) => {
    const store = createStore(combineReducers(reducers), state);
    return {
      ...store,
      persistor: {
        persist: () => null
      }
    };
};

describe('testing Shop component', () => {
    let wrapper;
    let mockFetchCollectionsStart;
    let store;

    beforeEach(() => {
        const mockReducer = (
            state = {
              isFetching: true
            },
            action
        ) => state;
      
        const mockState = {
            shop: {
              isFetching: true
            }
        };

        mockFetchCollectionsStart = jest.fn();

        store = createMockStore({
            state: mockState,
            reducers: { shop: mockReducer }
        });

        const mockMatch = {
            path: ''
        };

        const mockProps = {
            match: mockMatch,
            fetchCollectionsStart: mockFetchCollectionsStart
        };

        wrapper = mount(
            <BrowserRouter>
              <Provider store={store}>
                <Shop {...mockProps} />
              </Provider>
            </BrowserRouter>
        );
    });

    it('should render a Shop component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call a fetchCollectionStart', () => {
        expect(mockFetchCollectionsStart).toHaveBeenCalled();
    });
});