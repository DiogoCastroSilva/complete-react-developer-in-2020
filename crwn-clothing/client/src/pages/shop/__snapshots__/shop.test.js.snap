// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Shop component should render a Shop component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "persistor": Object {
            "persist": [Function],
          },
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Shop
        fetchCollectionsStart={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        match={
          Object {
            "path": "",
          }
        }
      >
        <div>
          <Suspense
            fallback={<Spinner />}
          >
            <Route
              exact={true}
              path=""
              render={[Function]}
            >
              <Component
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              />
            </Route>
            <Route
              path="/:collectionId"
              render={[Function]}
            />
            <Spinner>
              <styled.div>
                <div
                  className="sc-bdnylx gRtvuV"
                >
                  <styled.div>
                    <div
                      className="sc-gtssRu hnAUVI"
                    />
                  </styled.div>
                </div>
              </styled.div>
            </Spinner>
          </Suspense>
        </div>
      </Shop>
    </Provider>
  </Router>
</BrowserRouter>
`;
